src/vmm/src/arch/aarch64/cache_info.rs

```diff
@@ -298,22 +288,20 @@ pub(crate) fn read_cache_config(
     let mut logged_missing_attr = false;
     let engine = CacheEngine::default();
 
-    for index in 0..(MAX_CACHE_LEVEL + 1) {
+    for index in 0..=MAX_CACHE_LEVEL {
         match CacheEntry::from_index(index, engine.store.as_ref()) {
             Ok(cache) => {
                 append_cache_level(cache_l1, cache_non_l1, cache);
             }
+            // Missing cache level or type means not further search is necessary.
+            Err(Error::MissingCacheLevel) | Err(Error::MissingCacheType) => break,
             // Missing cache files is not necessary an error so we
-            // do not propagate it upwards. We were prudent enough to log a warning.
-            Err(Error::MissingCacheConfig) => return Ok(()),
+            // do not propagate it upwards. We were prudent enough to log it.
             Err(Error::MissingOptionalAttr(msg, cache)) => {
                 let level = cache.level;
                 append_cache_level(cache_l1, cache_non_l1, cache);
                 if !msg.is_empty() && !logged_missing_attr {
-                    warn!(
-                        "{}",
-                        format!("Could not read the {} for cache level {}.", msg, level)
-                    );
+                    warn!("Could not read the {} for cache level {}.", msg, level);
```

Comment:
Rather than logging the negative case, can we log only the positives? So instead of this:

```
WARN Could not read ... for cache level 4.
```

we have this in the logs:

```
INFO Found cache level 1
INFO Found cache level 2
INFO Found cache level 3
```

Reply:
This warning is about optional attributes and it is not an issue here. I think logging with `INFO` just to log is not a way. (no one really cares if caches are found successfully).

Reply:
If it's not important to log if we find a cache successfully or unsuccessfully, then why do we log at all?

Reply:
The main change in this PR is in these lines: https://github.com/firecracker-microvm/firecracker/blob/9ab568b4892836841a358126f02f2d947e86578b/src/vmm/src/arch/aarch64/cache_info.rs#L90
Previously we were emitting warnings when we could not read level of the cache, but then we were just ignoring the error, because it is not critical.

Reply:
And what are we doing now? It seems we are still logging a warning?

Reply:
The remaining warning is about missing attributes. I did not change the warning itself, just removed `format!` macro. I actually never seen this particular warning, but I don't mind converting it into `info!` as well.

Reply:
If we are making changes here, I think we should really consider not emitting a warning for something that is normal operations. One way may be to not log any warnings, at all. Another may be to log what caches we find, with Info or Debug.

Also include before/after logging in the commit so we understand what has changed.

Reply:
Replaced this particular warn and replaced it with `info` level logs.

Reply:
@kalyazin Added a more descriptive commit message.

Reply:
> @kalyazin Added a more descriptive commit message.

Apart from couple typos in the message ("cahce", "they was"),
> The error handing of such situations did not change.

this is a bit ambiguous. Maybe provide examples of the errors to make it clear that they are separate from the warnings that have been removed.

Reply:
@kalyazin Fixed the typos. Regarding examples of errors: there are no changes in error, the whole change is basically in just removing couple `warn!(...)` lines. I have no idea how to make it more clear.